spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -1172,7 +1172,28 @@ public B delay(String groupId, Consumer<DelayerEndpointSpec> endpointConfigurer)
 	 * @see EnricherSpec
 	 */
 	public B enrich(Consumer<EnricherSpec> enricherConfigurer) {
-		return register(new EnricherSpec(), enricherConfigurer);
+		EnricherSpec enricherSpec = new EnricherSpec();
+		enricherConfigurer.accept(enricherSpec);
+		Collection<Object> componentsToRegister = enricherSpec.getComponentsToRegister();
+		if (!CollectionUtils.isEmpty(componentsToRegister)) {
+			for (Object component : componentsToRegister) {
+				// Currently EnricherSpec only has IntegrationFlowBuilder as a sub component
+				if (component instanceof IntegrationFlowDefinition) {
+					IntegrationFlowDefinition<?> flowBuilder = (IntegrationFlowDefinition<?>) component;
+					if (flowBuilder.isOutputChannelRequired()) {
+						addComponent(flowBuilder.get());
+					}
+					else {
+						throw new BeanCreationException("Enricher subFlow must require an output channel");
```

Comment:
To be honest I don't see reason in this restriction.
There is no guarantee that flow ends with just a reply producer. It may sends to the channel and a flow behind it may return the result to the `replyChannel`.

Yes, that is the point to wait for the reply from the downstream flow for the `enricher`, but similar concern with unpredictable sub-flow behavior we have even with an existing XML definition.
So, my point to relax this assertion and really just configure `requestChannel` as an input to the that sub-flow and 
```
this.componentsToRegister.add(flowBuilder.get());
```
in the `EnricherSpec`.

That way this `IntegrationFlowDefinition` won't be affected at all.

The problem with "no reply from sub-flow" remains to the end-user. 
No choice!